mono/metadata/unity-memory-info.c

```diff
@@ -656,6 +641,10 @@ MonoManagedMemorySnapshot* mono_unity_capture_memory_snapshot()
 
 	g_hash_table_destroy(monoImages);
 
+	if (!wasDisabled)
+		GC_enable();
```

Comment:
Not sure it matters, but I notice you disable then stop and enable then start. Should those be mirrored, i.e. disable -> stop, start -> enable?

Reply:
yea it should be start->enable there. The reason I'm checking if we are already disabled is that the disable signal looks to be that it can be external to the capture running and so we should not modify it's state

Reply:
Yes, but a) I think it's ref counted. Will verify that b) checking this seems like a race condition

Reply:
removed all if's pertaining to the GC_enable/disable checks and waiting for the collect